File: src/03_problem_solving_techniques/05_divide_and_conquer/ts/divide_and_conquer.test.ts
import { sumArrayItems } from "./divide_and_conquer";

describe("SumArrayItems()", () => {
  test("Provides correct result", async () => {
    const array = [2, 4, 6];

    const result = sumArrayItems(array);

    expect(result).toBe(12);
  });
});

File: src/03_problem_solving_techniques/05_divide_and_conquer/rs/divide_and_conquer_test.rs
#[cfg(test)]
mod tests {
  use super::super::divide_and_conquer::sum_array_items;

  #[test]
  fn disjoint_set_is_instantiated() {
    let array = vec![2, 4, 6];
    let result = sum_array_items(&array);
    let expected = 12;

    assert_eq!(expected, result);
  }
}
